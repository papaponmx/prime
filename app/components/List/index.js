/**
 *
 * List
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
// import styled from 'styled-components';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

/* eslint-disable */
const List = ({ list, listType = null }) => {
  const prop = listType === null ? 'header' : `${listType}Header`;
  return (
    <div>
    <h3>
    <FormattedMessage {...messages[prop]} />
    </h3>
    <ul>
    {list.map(({id, name}) => <li key={id}>
      <Link to={`goals/${id}`}> {name} </Link>
      </li>
    )}
    </ul>
    </div>
  );
};
/* eslint-enable */

List.propTypes = {
  list: PropTypes.array.isRequired,
  listType: PropTypes.oneOf(['goals', 'roles', null]),
};

export default List;
