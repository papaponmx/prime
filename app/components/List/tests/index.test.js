import React from 'react';
import { shallow } from 'enzyme';
import { mockList as list } from './mocs';
import List from '../index';

describe('<List />', () => {
  let listType;
  let mounted;

  beforeEach(() => {
    mounted = shallow(
      List({
        list,
        listType,
      }),
    );
  });

  it('Expect to have unit tests specified', () => {
    expect(mounted).toMatchSnapshot();
  });

  xit('Should have a header listType as a fallback', () => {
    expect(mounted.props.listType).toBe('header');
  });

  it('Should render a header', () => {
    expect(mounted.find('h3')).toBeTruthy();
  });

  it('should render an <ul> tag', () => {
    const renderedComponent = shallow(<List list={list} />);
    expect(renderedComponent.type()).toEqual('div');
  });
});
