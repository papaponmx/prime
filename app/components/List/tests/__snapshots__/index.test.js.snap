// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Expect to have unit tests specified 1`] = `
<div>
  <h3>
    <FormattedMessage
      defaultMessage="This is the List component !"
      id="app.components.List.header"
      values={Object {}}
    />
  </h3>
  <ul>
    <li
      key="goal-.mzir08u"
    >
      <Link
        replace={false}
        to="goals/goal-.mzir08u"
      >
         
        Do something
         
      </Link>
    </li>
    <li
      key="goal-.ppalv8o"
    >
      <Link
        replace={false}
        to="goals/goal-.ppalv8o"
      >
         
        Some goal
         
      </Link>
    </li>
  </ul>
</div>
`;
